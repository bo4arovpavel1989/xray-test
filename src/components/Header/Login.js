import React from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import './Header.sass'

const Login = props => {
  const { login, authFail, isLogging } = props

  return (
      <form className='loginForm' onSubmit = { login }>
        {
          authFail ?
            <div className='authFailMessage'>Неверное имя пользователя или пароль!</div>
          :
            ''
        }
        <span>
          <input type='login' name='login' placeholder='login'/>
          <input type='password' name='password' placeholder='password'/>
        </span>
        <span>
          <input disabled = { isLogging } type='submit' value='Войти'/>
        </span>
      </form>
  )
}

Login.propTypes = {
  login: PropTypes.func.isRequired
}

export default withRouter(Login)
