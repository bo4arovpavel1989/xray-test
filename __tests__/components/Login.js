import React from 'react';
import { BrowserRouter } from 'react-router-dom'
import Login from '../../src/components/Header/Login.js';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Login component', () => {
  const props = {
    authFail: false,
    isLogging: false,
    login: jest.fn()
  };

  it('should render correct', () => {
      const login = mount(
        <BrowserRouter>
          <Login { ...props }/>
        </BrowserRouter>
      );

      expect(shallowToJson(login)).toMatchSnapshot()
  })

  it('should render corect when logging failed', () => {
    props.authFail = true;
    const login = mount(
      <BrowserRouter>
        <Login { ...props }/>
      </BrowserRouter>
    );

    expect(shallowToJson(login)).toMatchSnapshot()
  })

  it('should render disable submit when logging', () => {
    props.authFail = false;
    props.isLogging = true;
    const login = mount(
      <BrowserRouter>
        <Login { ...props }/>
      </BrowserRouter>
    );

    expect(shallowToJson(login)).toMatchSnapshot()
  })

  it('should call login function on submit', () => {
    props.authFail = false;
    props.isLogging = false;
    const login = mount(
      <BrowserRouter>
        <Login { ...props }/>
      </BrowserRouter>
    );

    login.find('form').simulate('submit');
    expect(props.login).toHaveBeenCalledTimes(1);
  })
})
