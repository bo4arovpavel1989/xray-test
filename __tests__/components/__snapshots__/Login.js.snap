// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render corect when logging failed 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Login)
      authFail={true}
      isLogging={false}
      login={[MockFunction]}
    >
      <Route>
        <Login
          authFail={true}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isLogging={false}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          login={[MockFunction]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <form
            className="loginForm"
            onSubmit={[MockFunction]}
          >
            <div
              className="authFailMessage"
            >
              Неверное имя пользователя или пароль!
            </div>
            <span>
              <input
                name="login"
                placeholder="login"
                type="login"
              />
              <input
                name="password"
                placeholder="password"
                type="password"
              />
            </span>
            <span>
              <input
                disabled={false}
                type="submit"
                value="Войти"
              />
            </span>
          </form>
        </Login>
      </Route>
    </withRouter(Login)>
  </Router>
</BrowserRouter>
`;

exports[`Login component should render correct 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Login)
      authFail={false}
      isLogging={false}
      login={[MockFunction]}
    >
      <Route>
        <Login
          authFail={false}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isLogging={false}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          login={[MockFunction]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <form
            className="loginForm"
            onSubmit={[MockFunction]}
          >
            <span>
              <input
                name="login"
                placeholder="login"
                type="login"
              />
              <input
                name="password"
                placeholder="password"
                type="password"
              />
            </span>
            <span>
              <input
                disabled={false}
                type="submit"
                value="Войти"
              />
            </span>
          </form>
        </Login>
      </Route>
    </withRouter(Login)>
  </Router>
</BrowserRouter>
`;

exports[`Login component should render disable submit when logging 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Login)
      authFail={false}
      isLogging={true}
      login={[MockFunction]}
    >
      <Route>
        <Login
          authFail={false}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isLogging={true}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          login={[MockFunction]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <form
            className="loginForm"
            onSubmit={[MockFunction]}
          >
            <span>
              <input
                name="login"
                placeholder="login"
                type="login"
              />
              <input
                name="password"
                placeholder="password"
                type="password"
              />
            </span>
            <span>
              <input
                disabled={true}
                type="submit"
                value="Войти"
              />
            </span>
          </form>
        </Login>
      </Route>
    </withRouter(Login)>
  </Router>
</BrowserRouter>
`;
